launcher/internal/oci/interface.go

```diff
@@ -0,0 +1,59 @@
+// Package oci contains functionalities to interact with OCI image signatures.
+// https://github.com/opencontainers/image-spec/tree/main#readme.
+package oci
+
+import (
+	"go.uber.org/multierr"
+)
+
+// SigningAlgorithm is a specific type for string constants used for sigature signing and verification.
+type SigningAlgorithm string
+
+const (
+	// RsassaPssSha256 is RSASSA-PSS with a SHA256 digest supported for cosign sign.
+	RsassaPssSha256 SigningAlgorithm = "RSASSA_PSS_SHA256"
+	// RsassaPkcs1v15Sha256 is RSASSA-PKCS1 v1.5 with a SHA256 digest supported for cosign sign.
+	RsassaPkcs1v15Sha256 SigningAlgorithm = "RSASSA_PKCS1V15_SHA256"
+	// EcdsaP256Sha256 is ECDSA on the P-256 Curve with a SHA256 digest supported for cosign sign.
+	EcdsaP256Sha256 SigningAlgorithm = "ECDSA_P256_SHA256"
```

Comment:
ED25519 is a possibility too.

Reply:
`ED25519` is not supported for the first launch of this feature as designed, see go/signed-image-impl-design